examples/resnet/resnet_main.py

```diff
@@ -0,0 +1,121 @@
+"""ResNet Train module, with some code from
+https://github.com/tensorflow/models/tree/master/resnet.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import cifar_input
+import numpy as np
+import resnet_model
+import ray
+import tensorflow as tf
+
+FLAGS = tf.app.flags.FLAGS
+tf.app.flags.DEFINE_string('train_data_path', '',
+                           'Filepattern for training data.')
+tf.app.flags.DEFINE_string('eval_data_path', '',
+                           'Filepattern for eval data')
+tf.app.flags.DEFINE_string('num_gpus', 1, 'Number of gpus to run with')
+
+@ray.remote
+def get_test(path):
+ os.environ['CUDA_VISIBLE_DEVICES'] = ''
+ with tf.device('/cpu:0'):
+  images, labels = cifar_input.build_input(path, 5000, False)
+  sess = tf.Session()
+  coord = tf.train.Coordinator()
+  tf.train.start_queue_runners(sess, coord=coord)
+  batches = [sess.run([images, labels]) for _ in range(2)]
+  coord.request_stop()
+  sess.close()
+  return (np.concatenate([batches[i][0] for i in range(2)]), np.concatenate([batches[i][1] for i in range(2)]))
+
+@ray.actor(num_gpus=1)
+class ResNetActor(object):
+  def __init__(self, path, num_gpus):
+    """Construct an actor that implements a residual network.
+
+    Args:
+      path: The path to the training data.
+      num_gpus: The number of ResNet actors that will be created to do data
+        parallel training. This value will be used to adjust the learning rate.
+    """
+    # Set the CUDA_VISIBLE_DEVICES environment variable in order to restrict
+    # which GPUs TensorFlow uses. Note that this only works if it is done before
+    # the call to tf.Session.
+    os.environ['CUDA_VISIBLE_DEVICES'] = ','.join([str(i) for i in ray.get_gpu_ids()])
+    hps = resnet_model.HParams(batch_size=128,
+                               num_classes=10,
+                               min_lrn_rate=0.0001,
+                               lrn_rate=0.1,
+                               num_residual_units=5,
+                               use_bottleneck=False,
+                               weight_decay_rate=0.0002,
+                               relu_leakiness=0.1,
+                               optimizer='mom',
+                               num_gpus=num_gpus)
+    with tf.Graph().as_default():
+      tf.set_random_seed(ray.get_gpu_ids()[0] + 1)
+      with tf.device('/gpu:0'):
+        images, labels = cifar_input.build_input(path, hps.batch_size, True)
```

Comment:
It'd be nice for this example to run easily on multiple machines, but that will require the Cifar data to be on each machine. Do you think it makes sense to put the data in the object store and pass the object IDs to the ResNet constructor? Or does that not make sense?